cmd/utils/cmd.go

```diff
@@ -287,6 +290,54 @@ func ExportAppendChain(blockchain *core.BlockChain, fn string, first uint64, las
 	return nil
 }
 
+func ExportHistory(bc *core.BlockChain, dir string, first, last, step uint64) error {
+	log.Info("Exporting blockchain history", "dir", dir)
+	if head := bc.CurrentBlock().NumberU64(); head < last {
+		log.Warn("Last block beyond head, setting last = head", "head", head, "last", last)
+		last = head
+	}
+	network := "unknown"
+	if name, ok := params.NetworkNames[bc.Config().ChainID.String()]; ok {
+		network = name
+	}
+	for i := uint64(0); i < last; i += step {
+		// Open file for Era.
+		fn := path.Join(dir, fmt.Sprintf("%s-%05d.era", network, i/step))
+		fh, err := os.OpenFile(fn, os.O_CREATE|os.O_APPEND|os.O_WRONLY, os.ModePerm)
+		if err != nil {
+			return err
+		}
+		defer fh.Close()
```

Comment:
`defer` inside a loop is usually not great, they will pile up and you will close `N` files when the whole thing is done and the method ends.

Reply:
Good point. Updated.